index.ts

```diff
@@ -0,0 +1,60 @@
+const products = require("./data/products.json");
+
+interface Product {
+  name: string;
+  description: string;
+  price: number;
+  image: URL;
+  limit: number;
+  amount: number;
+}
+
+interface Cart {
+  products: Product[];
+}
+
+const addProduct = (cart: Cart, product: Product) => {
+  cart.products.push(product);
+};
+
+const removeProduct = (cart: Cart, productToRemove: Product) => {
+  cart.products.forEach((product, i) => {
+    if (product.name === productToRemove.name) cart.products.splice(i, 1);
+  });
+};
+
+const updateProductAmount = (
+  cart: Cart,
+  productToChangeAmount: Product,
+  amount: number
+) => {
+  cart.products.forEach((product, i) => {
+    if (product.name === productToChangeAmount.name) product.amount = amount;
+  });
+};
+
+const checkout = (cart: Cart) => {
+  cart.products = [];
+};
+
+const totalPrice = (cart: Cart): number => {
+  let sum: number = 0;
+  cart.products.forEach((product, i) => {
+    sum += product.amount * product.price;
+  });
+  return sum;
+};
+
+const productQuantity = (cart: Cart) => {
```

Comment:
where is it used? if not used why is it needed?

Reply:
it is requested in the exercise but actually not used because the Cart properties aren't encapsulated and are accessible, so we can always access Cart.products.length